### ISSUE TITLE
File decoding fails sometimes due to repeated lines in file buffer

### ISSUE THREAD

**Issue by chrysplusplus** [chrys-8/subby#8]
With certain scenario files, the file buffer reaches a seemingly arbitrary line and then restarts from the beginning of the file, which naturally upsets the SRT decoder. 

After some investigation, this seems to be caused by the file cache not being reset when the file is read for a second time. This happens because the program attempts to decode the file with UTF-8, but falls back to Latin-1 encoding if there is a unicode decoding error.

TL;DR either file reading needs to be reset somehow within the current code architecture, or file encoding needs to be determined more intelligently (or passed in by the user, which for this is not realistic)

**Comment by chrysplusplus** [comment]
Due to the nature of the file that first brought this bug to my attention, I'm naming this Frankenstein's Bug.
